File: Medii de Proiectare si Programare/HW/mpp_hw/src/main/java/mpp/com/Domain/Child.java
package mpp.com.Domain;

import java.util.List;

public class Child extends Entity<Long>{
    private String name;
    private int age;
    private List<Track> tracks;
    private int numberOfTracks;

    public Child(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public Child(String name, int age, List<Track> tracks) {
        this.name = name;
        this.age = age;
        this.tracks = tracks;
        this.numberOfTracks = tracks.size();
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public List<Track> getTracks() {
        return tracks;
    }

    public void setTracks(List<Track> tracks) {
        this.tracks = tracks;
        this.numberOfTracks = tracks.size();
    }

    public int getNumberOfTracks() {
        return numberOfTracks;
    }
}

File: Medii de Proiectare si Programare/HW/MPP_CSharp/MPP_CSharp/Domain/Child.cs
using System.Collections.Generic;
using System;

namespace MPP_CSharp.Domain
{
    public class Child : Entity<long>
    {
        public string Name { get; set; }
        public int Age { get; set; }
        private List<Track> _tracks;

        public List<Track> Tracks { get; set; }
        public Child(long id, string name, int age) : base(id)
        {
            Name = name;
            Age = age;
        }
        
        public Child(string name, int age)
        {
            Name = name;
            Age = age;
        }

        public Child(string name, int age, List<Track> tracks)
        {
            Name = name;
            Age = age;
            Tracks = tracks;
        }

        public override bool Equals(object? obj)
        {
            if ( obj == null || GetType() != obj.GetType())
            {
                return false;
            }
            var child = (Child) obj;
            return Id == child.Id && Name == child.Name && Age == child.Age;
        }

        public override int GetHashCode()
        {
           return HashCode.Combine(Id, Name, Age);
        }
    }
}